internal/clients/oidc.go

```diff
@@ -0,0 +1,49 @@
+package clients
+
+import (
+	"context"
+
+	"github.com/coreos/go-oidc"
+	"github.com/ydataai/go-core/pkg/common/logging"
+	"golang.org/x/oauth2"
+)
+
+// OIDCClient defines a struct that can be used
+type OIDCClient struct {
+	Configuration OIDCConfiguration
+	OAuth2Config  *oauth2.Config
+	Verifier      *oidc.IDTokenVerifier
+	logger        logging.Logger
+}
+
+// NewOIDCClient defines a new values for the server
+func NewOIDCClient(logger logging.Logger, config OIDCConfiguration) *OIDCClient {
+	ctx := context.Background()
+
+	provider, err := oidc.NewProvider(ctx, config.OIDProviderURL)
```

Comment:
The instruction `oidc.NewProvider` will make an [http call](https://github.com/coreos/go-oidc/blob/6ce86d9e30589a68cbf891b7c4fc0d316201b786/oidc/oidc.go#L147), so, a `context.WithTimeout` should be used here instead of `context.Background`.  The timeout should be provided as a configuration parameter.

Reply:
I think we can pass the `context` as a function parameter also.

Reply:
In this case, we can't include context.WithTimeout, because the `oidc.NewProvider` returns a struct with context and when we do `defer cancel()`, we cancel all context:
```
return &Provider{
		issuer:       p.Issuer,
		authURL:      p.AuthURL,
		tokenURL:     p.TokenURL,
		userInfoURL:  p.UserInfoURL,
		algorithms:   algs,
		rawClaims:    body,
		remoteKeySet: NewRemoteKeySet(ctx, p.JWKSURL),
	}, nil
```

e.g:
`Failed to verify ID Token: failed to verify signature: fetching keys oidc: get keys failed Get "http://127.0.0.1:5556/keys": context canceled`

Reply:
I will 😢. I got it, that's a bad practice made by oidc library folks. As far as I know, store a context in a structure can cause this kind of side effect, once the context can be cancelled for someone in somewhere, we can have trouble.